pkg/drivers/kvm/kvm.go

```diff
@@ -443,16 +443,31 @@ func (d *Driver) Remove() error {
 	dom, err := conn.LookupDomainByName(d.MachineName)
 	if err != nil {
 		log.Warn("Domain %s does not exist, nothing to clean up...", d.MachineName)
+		return nil
 	}
-	if dom != nil {
-		log.Infof("Domain %s exists, removing...", d.MachineName)
-		if err := dom.Destroy(); err != nil {
-			return err
-		}
-		if err := dom.Undefine(); err != nil {
-			return err
-		}
+
+	log.Infof("Domain %s exists, removing...", d.MachineName)
+	if err := d.destroyRunningDomain(dom); err != nil {
+		return errors.Wrap(err, "destroying running domain")
+	}
+
+	if err := dom.Undefine(); err != nil {
+		return err
 	}
 
 	return nil
 }
+
+func (d *Driver) destroyRunningDomain(dom *libvirt.Domain) error {
+	state, reason, err := dom.GetState()
+	if err != nil {
+		return err
```

Comment:
Since this function returns two types of errors, do you mind using errors.Wrap() on at least one of them so we can tell from the error message which function caused the error to occur?

Reply:
sure!